docs/specification/draft/client/elicitation.mdx

```diff
@@ -317,16 +521,199 @@ The three response actions are:
+- **Decline**: Handle explicit rejection (e.g., offer alternatives)
 - **Cancel**: Handle dismissal (e.g., prompt again later)
 
+## Implementation Considerations
+
+### Statefulness
+
+Most practical uses of elicitation require that the server maintain state about users:
+
+- Whether required information has been collected (e.g., the user's display name via form elicitation)
+- Status of resource access (e.g., API keys or a payment flow via out-of-band elicitation)
+
+Servers implementing elicitation **MUST** securely associate this state with individual users following the guidelines in the [security best practices](../basic/security_best_practices) document. Specifically:
+
+- State **MUST NOT** be associated with session IDs alone
+- User identification **MUST** be derived from authenticated tokens
+- State storage **MUST** be protected against unauthorized access
+
+<Note>
+  The examples in this section are non-normative and illustrate potential uses
+  of elicitation. Implementers should adapt these patterns to their specific
+  requirements while maintaining security best practices.
+</Note>
+
+### Out-of-Band Elicitation for Sensitive Data
+
+For servers that interact with APIs requiring sensitive credentials (e.g. LLM APIs), out-of-band elicitation provides a secure mechanism for collecting API keys without exposing them to the MCP client or host application.
+
+In this pattern:
+
+1. The server directs users to a secure web page (served over HTTPS)
+2. The page presents a branded form UI on a domain the user trusts
+3. Users enter sensitive credentials directly into the secure form
+4. The server stores credentials securely, bound to the user's identity
+5. Subsequent MCP requests use these stored credentials for API access
+
+This approach ensures that sensitive credentials never pass through the MCP client or any intermediate MCP servers, reducing the risk of exposure through client-side logging or other attack vectors.
+
+### Out-of-Band Elicitation for OAuth Flows
+
+Out-of-band elicitation enables a pattern where MCP servers act as OAuth clients to third-party resource servers.
+The "downstream authorization" enabled by out-of-band elicitation is fundamentally separate from [MCP authorization](../basic/authorization).
+
+<Note>
+  This distinction is critical to understand. Out-of-band elicitation is *not*
+  about authorizing (or re-authorizing) the MCP client to access the MCP server.
+  It *can* be used to enable the MCP server to access third-party resources or
+  APIs on behalf of the user, without exposing sensitive credentials or tokens
+  to the MCP client.
+</Note>
+
+#### Understanding the Distinction
+
+- **MCP Authorization**: Required OAuth flow between the MCP client and MCP server (covered in the [authorization specification](../basic/authorization))
+- **Downstream Authorization**: Optional OAuth flow between the MCP server and a third-party resource server, initiated via out-of-band elicitation
```

Comment:
```suggestion
- **Downstream Authorization**: Optional authorization between the MCP server and a third-party resource server, initiated via out-of-band elicitation
```

It sounds like one of the use cases here is obtaining API keys (like GitHub PAT) so the authorization between the MCP server and the RS is not always OAuth.